server/integrationtests/board_test.go

```diff
@@ -150,6 +150,28 @@ func TestCreateBoard(t *testing.T) {
 			require.Equal(t, board.ID, members[0].BoardID)
 			require.True(t, members[0].SchemeAdmin)
 		})
+
+		t.Run("creator should be able to access the public board and its blocks", func(t *testing.T) {
+			rbBoard, resp := th.Client.GetBoard(board.ID, "")
+			th.CheckOK(resp)
+			require.NotNil(t, rbBoard)
+			require.Equal(t, board, rbBoard)
+
+			rBlocks, resp := th.Client.GetBlocksForBoard(board.ID)
+			th.CheckOK(resp)
+			require.NotNil(t, rBlocks)
+		})
+
+		t.Run("another user should be able to access the public board and its blocks", func(t *testing.T) {
+			rbBoard, resp := th.Client2.GetBoard(board.ID, "")
+			th.CheckOK(resp)
+			require.NotNil(t, rbBoard)
+			require.Equal(t, board, rbBoard)
+
+			rBlocks, resp := th.Client2.GetBlocksForBoard(board.ID)
+			th.CheckOK(resp)
```

Comment:
This test is currently failing. The GetBlocks API is not checking for team access on public  boards.